src/game/driver.py

```diff
@@ -0,0 +1,46 @@
+import threading
+
+
+class Driver:
+    def __init__(self, game):
+        self.game = game
+
+    def start_game(self):
+        self.game.start_game()
+        self._tick()
+
+    def re_tick(self):
+        raise NotImplementedError()
+
+    def _tick(self):
+        if not self.game.is_over():
+            self.game.single_tick()
+            self.re_tick()
+        elif self.game.is_over() and not self.game.players_alive():
```

Comment:
I'd move that logic to the game. How about `single_tick` returns a `GameState`? That way the driver knows when to stop, but it doesn't have to handle the logic of calling `tied_game` or `game_won`